server/xcode/xcode_darwin.go

```diff
@@ -90,6 +96,14 @@ func (x *xcodeLocator) locate() error {
 	urlRefs := make([]C.CFURLRef, n)
 	C.CFArrayGetValues(urlsArrayRef, C.CFRange{0, n}, (*unsafe.Pointer)(unsafe.Pointer(&urlRefs[0])))
 
+	var defaultDeveloperDir string
+	out, err := exec.Command("xcode-select", "-p").Output()
+	if err != nil {
+		log.Warningf("Error calling xcode-select: %s", err.Error())
+	} else {
+		defaultDeveloperDir = strings.TrimSpace(string(out))
+	}
+
```

Comment:
nit: kind of a matter of taste, but given how long "locate()" is I think we should put this in a a function:

// xcodeSelectDir returns the active developer dir according to xcode-select or
// the empty string if xcode select did not run.
func xcodeSelectDir() string {
  out, err := exec.Command("xcode-select", "-p").Output()
  if err != nil {
		log.Warningf("Error calling xcode-select: %s", err.Error())
        return ""
  }
  return strings.TrimSpace(string(out))
}


...

defaultDeveloperDir = x.xcodeSelectDir()